filepathfilter/filepathfilter_test.go

```diff
@@ -55,34 +50,37 @@ func TestPatternMatch(t *testing.T) {
 	assertPatternMatch(t, "sub",
 		"sub/",
 		"sub",
-		"sub/filename.txt",
 		"top/sub/",
 		"top/sub",
+	)
+	refutePatternMatch(t, "sub",
+		"sub/filename.txt",
 		"top/sub/filename.txt",
 	)
 
-	assertPatternMatch(t, "sub/", "sub/filename.txt", "top/sub/filename.txt")
-	assertPatternMatch(t, "/sub", "sub/", "sub", "sub/filename.txt")
-	assertPatternMatch(t, "/sub/", "sub/filename.txt")
+	refutePatternMatch(t, "sub/", "sub/filename.txt", "top/sub/filename.txt")
+	assertPatternMatch(t, "/sub", "sub")
+	refutePatternMatch(t, "/sub", "sub/filename.txt")
+	refutePatternMatch(t, "/sub/", "sub/filename.txt")
 	refutePatternMatch(t, "/sub", "subfilename.txt", "top/sub/", "top/sub", "top/sub/filename.txt")
 	refutePatternMatch(t, "sub", "subfilename.txt")
 	refutePatternMatch(t, "sub/", "subfilename.txt")
 	refutePatternMatch(t, "/sub/", "subfilename.txt", "top/sub/filename.txt")
 
 	// nested path
-	assertPatternMatch(t, "top/sub",
+	assertPatternMatch(t, "top/sub", "top/sub")
+	refutePatternMatch(t, "top/sub",
 		"top/sub/filename.txt",
-		"top/sub/",
-		"top/sub",
-		"root/top/sub/filename.txt",
 		"root/top/sub/",
 		"root/top/sub",
+		"root/top/sub/filename.txt",
 	)
-	assertPatternMatch(t, "top/sub/", "top/sub/filename.txt")
-	assertPatternMatch(t, "top/sub/", "root/top/sub/filename.txt")
+	refutePatternMatch(t, "top/sub/", "top/sub/filename.txt")
+	refutePatternMatch(t, "top/sub/", "root/top/sub/filename.txt")
 
-	assertPatternMatch(t, "/top/sub", "top/sub/", "top/sub", "top/sub/filename.txt")
-	assertPatternMatch(t, "/top/sub/", "top/sub/filename.txt")
+	assertPatternMatch(t, "/top/sub", "top/sub")
+	refutePatternMatch(t, "/top/sub", "top/sub/filename.txt")
+	refutePatternMatch(t, "/top/sub/", "top/sub/filename.txt")
```

Comment:
Again, just as a though, what would you think about making this whole "nested path" [section](https://github.com/git-lfs/git-lfs/blob/48ffab044a182a24e515c4449b74b713cab1e2d5/filepathfilter/filepathfilter_test.go#L72-L99) more explicit and consistently organized, something like:
```go
	// nested path
	assertPatternMatch(t, "top/sub",
		"top/sub/",
		"top/sub",
	)
	refutePatternMatch(t, "top/sub",
		"top/subfilename.txt",
		"top/sub/filename.txt",
		"root/top/sub/",
		"root/top/sub",
		"root/top/sub/filename.txt",
	)

	assertPatternMatch(t, "/top/sub",
		"top/sub/",
		"top/sub",
	)
	refutePatternMatch(t, "/top/sub",
		"top/subfilename.txt",
		"top/sub/filename.txt",
		"root/top/sub/",
		"root/top/sub",
		"root/top/sub/filename.txt",
	)

	assertPatternMatch(t, "top/sub/",
		"top/sub/",
		"top/sub",
	)
	refutePatternMatch(t, "top/sub/",
		"top/subfilename.txt",
		"top/sub/filename.txt",
		"root/top/sub/",
		"root/top/sub",
		"root/top/sub/filename.txt",
	)

	assertPatternMatch(t, "/top/sub/",
		"top/sub/",
		"top/sub",
	)
	refutePatternMatch(t, "/top/sub/",
		"top/subfilename.txt",
		"top/sub/filename.txt",
		"root/top/sub/",
		"root/top/sub",
		"root/top/sub/filename.txt",
	)
```